controllers/awsmachine_controller.go

```diff
@@ -1141,3 +1149,19 @@ func (r *AWSMachineReconciler) ensureInstanceMetadataOptions(ec2svc services.EC2
 
 	return ec2svc.ModifyInstanceMetadataOptions(instance.ID, machine.Spec.InstanceMetadataOptions)
 }
+
+func (r *AWSMachineReconciler) getControlPlane(ctx context.Context, log *logger.Logger, cluster *clusterv1.Cluster) (*unstructured.Unstructured, error) {
+	controlPlane := unstructured.Unstructured{}
```

Comment:
```suggestion
	controlPlane := new(unstructured.Unstructured)
```
So we carry the pointer around?

Reply:
I think you can use `Get()` func from [sigs.k8s.io/cluster-api/controllers/external](https://github.com/kubernetes-sigs/cluster-api/blob/3a373971a953cdd1db97cdba89db5fba00452041/controllers/external/util.go#L34) pkg here

Reply:
Thanks, @muraee, implemented with bc6370bdb.